contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -2389,6 +2415,11 @@ static void bbf_ProcessUtility(PlannedStmt *pstmt,
 												   makeDefElem("original_user_name",
 															   (Node *) makeString(orig_user_name),
 															   -1));
+							/*
+							 * if the user name contains '\', check whether the login is from windows or not
+							 */
+							if (!from_windows)
+								validateUserAndRole(orig_user_name);
```

Comment:
this can be moved to 2523 with separate condition.

Reply:
Done!